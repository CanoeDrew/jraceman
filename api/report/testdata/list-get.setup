# The list-reports api command doesn't use anything from the database,
# so we don't need to create any data there.
