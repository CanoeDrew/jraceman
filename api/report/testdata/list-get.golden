[
  {
    "Name": "site-all-report",
    "Display": "Sites",
    "Description": "List all sites",
    "OrderBy": [
      {
        "Name": "name",
        "Display": "Site name"
      },
      {
        "Name": "city",
        "Display": "City"
      }
    ],
    "Where": []
  },
  {
    "Name": "org.jimmc.jraceman.Entries",
    "Display": "Entries",
    "Description": "Entries ordered as selected.",
    "OrderBy": [
      {
        "Name": "team",
        "Display": "Team, Person, Event"
      },
      {
        "Name": "person",
        "Display": "Person, Event"
      },
      {
        "Name": "eventTeam",
        "Display": "Event, Team, Person"
      },
      {
        "Name": "eventPerson",
        "Display": "Event, Person"
      }
    ],
    "Where": [
      {
        "Name": "event_id",
        "Display": "Event",
        "Ops": [
          "eq"
        ],
        "KeyTable": "event"
      },
      {
        "Name": "event_name",
        "Display": "Event Name",
        "Ops": [
          "eq",
          "ne",
          "gt",
          "ge",
          "lt",
          "le",
          "like"
        ],
        "KeyTable": ""
      },
      {
        "Name": "event_number",
        "Display": "Event Number",
        "Ops": [
          "eq",
          "ne",
          "gt",
          "ge",
          "lt",
          "le"
        ],
        "KeyTable": ""
      },
      {
        "Name": "team_id",
        "Display": "Team",
        "Ops": [
          "eq"
        ],
        "KeyTable": "team"
      },
      {
        "Name": "team_shortname",
        "Display": "Team Short Name",
        "Ops": [
          "eq",
          "ne",
          "gt",
          "ge",
          "lt",
          "le",
          "like"
        ],
        "KeyTable": ""
      },
      {
        "Name": "team_name",
        "Display": "Team Name",
        "Ops": [
          "eq",
          "ne",
          "gt",
          "ge",
          "lt",
          "le",
          "like"
        ],
        "KeyTable": ""
      },
      {
        "Name": "person_id",
        "Display": "Person",
        "Ops": [
          "eq"
        ],
        "KeyTable": "person"
      }
    ]
  },
  {
    "Name": "org.jimmc.jraceman.EntryCountPerEvent",
    "Display": "Entry Count Per Event",
    "Description": "The number of Entries and Teams entered for each Event.",
    "OrderBy": [
      {
        "Name": "event",
        "Display": "Event"
      },
      {
        "Name": "teamcount",
        "Display": "#Teams"
      },
      {
        "Name": "entrycount",
        "Display": "#Entries"
      },
      {
        "Name": "groupcount",
        "Display": "#Groups"
      }
    ],
    "Where": [
      {
        "Name": "event_id",
        "Display": "Event",
        "Ops": [
          "eq"
        ],
        "KeyTable": "event"
      },
      {
        "Name": "event_name",
        "Display": "Event Name",
        "Ops": [
          "eq",
          "ne",
          "gt",
          "ge",
          "lt",
          "le",
          "like"
        ],
        "KeyTable": ""
      },
      {
        "Name": "event_number",
        "Display": "Event Number",
        "Ops": [
          "eq",
          "ne",
          "gt",
          "ge",
          "lt",
          "le"
        ],
        "KeyTable": ""
      },
      {
        "Name": "meet_id",
        "Display": "Meet",
        "Ops": [
          "eq"
        ],
        "KeyTable": "meet"
      },
      {
        "Name": "person_id",
        "Display": "Person",
        "Ops": [
          "eq"
        ],
        "KeyTable": "person"
      },
      {
        "Name": "team_id",
        "Display": "Team",
        "Ops": [
          "eq"
        ],
        "KeyTable": "team"
      },
      {
        "Name": "team_shortname",
        "Display": "Team Short Name",
        "Ops": [
          "eq",
          "ne",
          "gt",
          "ge",
          "lt",
          "le",
          "like"
        ],
        "KeyTable": ""
      },
      {
        "Name": "team_name",
        "Display": "Team Name",
        "Ops": [
          "eq",
          "ne",
          "gt",
          "ge",
          "lt",
          "le",
          "like"
        ],
        "KeyTable": ""
      }
    ]
  },
  {
    "Name": "org.jimmc.jraceman.EntryCountPerEventNS",
    "Display": "Entry Count Per Event non-scoring",
    "Description": "The number of Entries and Teams entered, including non-scoring counts, for each Event.",
    "OrderBy": [
      {
        "Name": "event",
        "Display": "Event"
      },
      {
        "Name": "teamcount",
        "Display": "#Teams"
      },
      {
        "Name": "entrycount",
        "Display": "#Entries"
      },
      {
        "Name": "nsentries",
        "Display": "#NS-Entries"
      },
      {
        "Name": "groupcount",
        "Display": "#Groups"
      }
    ],
    "Where": [
      {
        "Name": "event_id",
        "Display": "Event",
        "Ops": [
          "eq"
        ],
        "KeyTable": "event"
      },
      {
        "Name": "event_name",
        "Display": "Event Name",
        "Ops": [
          "eq",
          "ne",
          "gt",
          "ge",
          "lt",
          "le",
          "like"
        ],
        "KeyTable": ""
      },
      {
        "Name": "event_number",
        "Display": "Event Number",
        "Ops": [
          "eq",
          "ne",
          "gt",
          "ge",
          "lt",
          "le"
        ],
        "KeyTable": ""
      },
      {
        "Name": "meet_id",
        "Display": "Meet",
        "Ops": [
          "eq"
        ],
        "KeyTable": "meet"
      },
      {
        "Name": "person_id",
        "Display": "Person",
        "Ops": [
          "eq"
        ],
        "KeyTable": "person"
      },
      {
        "Name": "team_id",
        "Display": "Team",
        "Ops": [
          "eq"
        ],
        "KeyTable": "team"
      },
      {
        "Name": "team_shortname",
        "Display": "Team Short Name",
        "Ops": [
          "eq",
          "ne",
          "gt",
          "ge",
          "lt",
          "le",
          "like"
        ],
        "KeyTable": ""
      },
      {
        "Name": "team_name",
        "Display": "Team Name",
        "Ops": [
          "eq",
          "ne",
          "gt",
          "ge",
          "lt",
          "le",
          "like"
        ],
        "KeyTable": ""
      }
    ]
  },
  {
    "Name": "org.jimmc.jraceman.EntryCountPerLevel",
    "Display": "Entry Count Per Level",
    "Description": "The number of Entries and Teams entered for each Level.",
    "OrderBy": [
      {
        "Name": "level",
        "Display": "Level"
      },
      {
        "Name": "teamcount",
        "Display": "#Teams"
      },
      {
        "Name": "personcount",
        "Display": "#People"
      },
      {
        "Name": "entrycount",
        "Display": "#Entries"
      },
      {
        "Name": "groupcount",
        "Display": "#Groups"
      }
    ],
    "Where": [
      {
        "Name": "event_id",
        "Display": "Event",
        "Ops": [
          "eq"
        ],
        "KeyTable": "event"
      },
      {
        "Name": "event_name",
        "Display": "Event Name",
        "Ops": [
          "eq",
          "ne",
          "gt",
          "ge",
          "lt",
          "le",
          "like"
        ],
        "KeyTable": ""
      },
      {
        "Name": "event_number",
        "Display": "Event Number",
        "Ops": [
          "eq",
          "ne",
          "gt",
          "ge",
          "lt",
          "le"
        ],
        "KeyTable": ""
      },
      {
        "Name": "meet_id",
        "Display": "Meet",
        "Ops": [
          "eq"
        ],
        "KeyTable": "meet"
      },
      {
        "Name": "person_id",
        "Display": "Person",
        "Ops": [
          "eq"
        ],
        "KeyTable": "person"
      },
      {
        "Name": "team_id",
        "Display": "Team",
        "Ops": [
          "eq"
        ],
        "KeyTable": "team"
      },
      {
        "Name": "team_shortname",
        "Display": "Team Short Name",
        "Ops": [
          "eq",
          "ne",
          "gt",
          "ge",
          "lt",
          "le",
          "like"
        ],
        "KeyTable": ""
      },
      {
        "Name": "team_name",
        "Display": "Team Name",
        "Ops": [
          "eq",
          "ne",
          "gt",
          "ge",
          "lt",
          "le",
          "like"
        ],
        "KeyTable": ""
      }
    ]
  },
  {
    "Name": "org.jimmc.jraceman.EntryCountPerPerson",
    "Display": "Entry Count Per Person",
    "Description": "The number of Entries entered for each Person.",
    "OrderBy": [
      {
        "Name": "person",
        "Display": "Person"
      },
      {
        "Name": "entrycount",
        "Display": "#Entries"
      }
    ],
    "Where": [
      {
        "Name": "event_id",
        "Display": "Event",
        "Ops": [
          "eq"
        ],
        "KeyTable": "event"
      },
      {
        "Name": "event_name",
        "Display": "Event Name",
        "Ops": [
          "eq",
          "ne",
          "gt",
          "ge",
          "lt",
          "le",
          "like"
        ],
        "KeyTable": ""
      },
      {
        "Name": "event_number",
        "Display": "Event Number",
        "Ops": [
          "eq",
          "ne",
          "gt",
          "ge",
          "lt",
          "le"
        ],
        "KeyTable": ""
      },
      {
        "Name": "meet_id",
        "Display": "Meet",
        "Ops": [
          "eq"
        ],
        "KeyTable": "meet"
      },
      {
        "Name": "person_id",
        "Display": "Person",
        "Ops": [
          "eq"
        ],
        "KeyTable": "person"
      },
      {
        "Name": "team_id",
        "Display": "Team",
        "Ops": [
          "eq"
        ],
        "KeyTable": "team"
      },
      {
        "Name": "team_shortname",
        "Display": "Team Short Name",
        "Ops": [
          "eq",
          "ne",
          "gt",
          "ge",
          "lt",
          "le",
          "like"
        ],
        "KeyTable": ""
      },
      {
        "Name": "team_name",
        "Display": "Team Name",
        "Ops": [
          "eq",
          "ne",
          "gt",
          "ge",
          "lt",
          "le",
          "like"
        ],
        "KeyTable": ""
      }
    ]
  },
  {
    "Name": "org.jimmc.jraceman.EntryCountPerTeam",
    "Display": "Entry Count Per Team",
    "Description": "The number of Entries for each Team.",
    "OrderBy": [
      {
        "Name": "team",
        "Display": "Team"
      },
      {
        "Name": "personcount",
        "Display": "#People"
      },
      {
        "Name": "entrycount",
        "Display": "#Entries"
      },
      {
        "Name": "groupcount",
        "Display": "#Groups"
      }
    ],
    "Where": [
      {
        "Name": "event_id",
        "Display": "Event",
        "Ops": [
          "eq"
        ],
        "KeyTable": "event"
      },
      {
        "Name": "event_name",
        "Display": "Event Name",
        "Ops": [
          "eq",
          "ne",
          "gt",
          "ge",
          "lt",
          "le",
          "like"
        ],
        "KeyTable": ""
      },
      {
        "Name": "event_number",
        "Display": "Event Number",
        "Ops": [
          "eq",
          "ne",
          "gt",
          "ge",
          "lt",
          "le"
        ],
        "KeyTable": ""
      },
      {
        "Name": "meet_id",
        "Display": "Meet",
        "Ops": [
          "eq"
        ],
        "KeyTable": "meet"
      },
      {
        "Name": "person_id",
        "Display": "Person",
        "Ops": [
          "eq"
        ],
        "KeyTable": "person"
      },
      {
        "Name": "team_id",
        "Display": "Team",
        "Ops": [
          "eq"
        ],
        "KeyTable": "team"
      },
      {
        "Name": "team_shortname",
        "Display": "Team Short Name",
        "Ops": [
          "eq",
          "ne",
          "gt",
          "ge",
          "lt",
          "le",
          "like"
        ],
        "KeyTable": ""
      },
      {
        "Name": "team_name",
        "Display": "Team Name",
        "Ops": [
          "eq",
          "ne",
          "gt",
          "ge",
          "lt",
          "le",
          "like"
        ],
        "KeyTable": ""
      }
    ]
  },
  {
    "Name": "org.jimmc.jraceman.LaneAssignments",
    "Display": "Lane Assignments",
    "Description": "Lane Assignments.",
    "OrderBy": [
      {
        "Name": "event",
        "Display": "Event, Race, Lane, Person"
      },
      {
        "Name": "race",
        "Display": "Race, Lane, Person"
      },
      {
        "Name": "teamPerson",
        "Display": "Team, Person"
      },
      {
        "Name": "teamRace",
        "Display": "Team, Race, Person"
      },
      {
        "Name": "person",
        "Display": "Person, Race"
      }
    ],
    "Where": [
      {
        "Name": "event_id",
        "Display": "Event",
        "Ops": [
          "eq"
        ],
        "KeyTable": "event"
      },
      {
        "Name": "event_name",
        "Display": "Event Name",
        "Ops": [
          "eq",
          "ne",
          "gt",
          "ge",
          "lt",
          "le",
          "like"
        ],
        "KeyTable": ""
      },
      {
        "Name": "event_number",
        "Display": "Event Number",
        "Ops": [
          "eq",
          "ne",
          "gt",
          "ge",
          "lt",
          "le"
        ],
        "KeyTable": ""
      },
      {
        "Name": "race_id",
        "Display": "Race",
        "Ops": [
          "eq"
        ],
        "KeyTable": "race"
      },
      {
        "Name": "race_number",
        "Display": "Race Number",
        "Ops": [
          "eq",
          "ne",
          "gt",
          "ge",
          "lt",
          "le"
        ],
        "KeyTable": ""
      },
      {
        "Name": "race_round",
        "Display": "Race Round",
        "Ops": [
          "eq",
          "ne",
          "gt",
          "ge",
          "lt",
          "le"
        ],
        "KeyTable": ""
      },
      {
        "Name": "race_start",
        "Display": "Race Start",
        "Ops": [
          "eq",
          "ne",
          "gt",
          "ge",
          "lt",
          "le",
          "like"
        ],
        "KeyTable": ""
      },
      {
        "Name": "team_id",
        "Display": "Team",
        "Ops": [
          "eq"
        ],
        "KeyTable": "team"
      },
      {
        "Name": "team_shortname",
        "Display": "Team Short Name",
        "Ops": [
          "eq",
          "ne",
          "gt",
          "ge",
          "lt",
          "le",
          "like"
        ],
        "KeyTable": ""
      },
      {
        "Name": "team_name",
        "Display": "Team Name",
        "Ops": [
          "eq",
          "ne",
          "gt",
          "ge",
          "lt",
          "le",
          "like"
        ],
        "KeyTable": ""
      },
      {
        "Name": "person_id",
        "Display": "Person",
        "Ops": [
          "eq"
        ],
        "KeyTable": "person"
      },
      {
        "Name": "event_race_entry_scratched",
        "Display": "Scratched",
        "Ops": [
          "eq",
          "ne",
          "gt",
          "ge",
          "lt",
          "le"
        ],
        "KeyTable": ""
      }
    ]
  },
  {
    "Name": "org.jimmc.jraceman.MedalCount",
    "Display": "Medal Count",
    "Description": "The number of medals received by each team",
    "OrderBy": [
      {
        "Name": "team",
        "Display": "Team"
      }
    ],
    "Where": [
      {
        "Name": "meet_id",
        "Display": "Meet",
        "Ops": [
          "eq"
        ],
        "KeyTable": "meet"
      },
      {
        "Name": "event_id",
        "Display": "Event",
        "Ops": [
          "eq"
        ],
        "KeyTable": "event"
      },
      {
        "Name": "event_name",
        "Display": "Event Name",
        "Ops": [
          "eq",
          "ne",
          "gt",
          "ge",
          "lt",
          "le",
          "like"
        ],
        "KeyTable": ""
      },
      {
        "Name": "event_number",
        "Display": "Event Number",
        "Ops": [
          "eq",
          "ne",
          "gt",
          "ge",
          "lt",
          "le"
        ],
        "KeyTable": ""
      },
      {
        "Name": "team_id",
        "Display": "Team",
        "Ops": [
          "eq"
        ],
        "KeyTable": "team"
      },
      {
        "Name": "team_shortname",
        "Display": "Team Short Name",
        "Ops": [
          "eq",
          "ne",
          "gt",
          "ge",
          "lt",
          "le",
          "like"
        ],
        "KeyTable": ""
      },
      {
        "Name": "team_name",
        "Display": "Team Name",
        "Ops": [
          "eq",
          "ne",
          "gt",
          "ge",
          "lt",
          "le",
          "like"
        ],
        "KeyTable": ""
      },
      {
        "Name": "person_id",
        "Display": "Person",
        "Ops": [
          "eq"
        ],
        "KeyTable": "person"
      }
    ]
  },
  {
    "Name": "org.jimmc.jraceman.PersonalResults",
    "Display": "Personal Results",
    "Description": "Personal results for each person collected on one page per person, sorted by team and last name",
    "OrderBy": [
      {
        "Name": "team",
        "Display": "Team"
      },
      {
        "Name": "person",
        "Display": "Person"
      }
    ],
    "Where": [
      {
        "Name": "meet_id",
        "Display": "Meet",
        "Ops": [
          "eq"
        ],
        "KeyTable": "meet"
      },
      {
        "Name": "team_id",
        "Display": "Team",
        "Ops": [
          "eq"
        ],
        "KeyTable": "team"
      },
      {
        "Name": "team_shortname",
        "Display": "Team Short Name",
        "Ops": [
          "eq",
          "ne",
          "gt",
          "ge",
          "lt",
          "le",
          "like"
        ],
        "KeyTable": ""
      },
      {
        "Name": "team_name",
        "Display": "Team Name",
        "Ops": [
          "eq",
          "ne",
          "gt",
          "ge",
          "lt",
          "le",
          "like"
        ],
        "KeyTable": ""
      },
      {
        "Name": "person_id",
        "Display": "Person",
        "Ops": [
          "eq"
        ],
        "KeyTable": "person"
      }
    ]
  },
  {
    "Name": "org.jimmc.jraceman.RaceLanes",
    "Display": "Race Lanes",
    "Description": "Lane assignments for a Race",
    "OrderBy": [
      {
        "Name": "event",
        "Display": "Event, Race"
      },
      {
        "Name": "race",
        "Display": "Race"
      }
    ],
    "Where": [
      {
        "Name": "event_id",
        "Display": "Event",
        "Ops": [
          "eq"
        ],
        "KeyTable": "event"
      },
      {
        "Name": "event_name",
        "Display": "Event Name",
        "Ops": [
          "eq",
          "ne",
          "gt",
          "ge",
          "lt",
          "le",
          "like"
        ],
        "KeyTable": ""
      },
      {
        "Name": "event_number",
        "Display": "Event Number",
        "Ops": [
          "eq",
          "ne",
          "gt",
          "ge",
          "lt",
          "le"
        ],
        "KeyTable": ""
      },
      {
        "Name": "race_id",
        "Display": "Race",
        "Ops": [
          "eq"
        ],
        "KeyTable": "race"
      },
      {
        "Name": "race_number",
        "Display": "Race Number",
        "Ops": [
          "eq",
          "ne",
          "gt",
          "ge",
          "lt",
          "le"
        ],
        "KeyTable": ""
      },
      {
        "Name": "race_round",
        "Display": "Race Round",
        "Ops": [
          "eq",
          "ne",
          "gt",
          "ge",
          "lt",
          "le"
        ],
        "KeyTable": ""
      },
      {
        "Name": "race_start",
        "Display": "Race Start",
        "Ops": [
          "eq",
          "ne",
          "gt",
          "ge",
          "lt",
          "le",
          "like"
        ],
        "KeyTable": ""
      }
    ]
  },
  {
    "Name": "org.jimmc.jraceman.RaceResults",
    "Display": "Race Results",
    "Description": "Results for a Race, sorted by finish order",
    "OrderBy": [
      {
        "Name": "event",
        "Display": "Event, Race"
      },
      {
        "Name": "race",
        "Display": "Race"
      }
    ],
    "Where": [
      {
        "Name": "event_id",
        "Display": "Event",
        "Ops": [
          "eq"
        ],
        "KeyTable": "event"
      },
      {
        "Name": "event_name",
        "Display": "Event Name",
        "Ops": [
          "eq",
          "ne",
          "gt",
          "ge",
          "lt",
          "le",
          "like"
        ],
        "KeyTable": ""
      },
      {
        "Name": "event_number",
        "Display": "Event Number",
        "Ops": [
          "eq",
          "ne",
          "gt",
          "ge",
          "lt",
          "le"
        ],
        "KeyTable": ""
      },
      {
        "Name": "race_id",
        "Display": "Race",
        "Ops": [
          "eq"
        ],
        "KeyTable": "race"
      },
      {
        "Name": "race_number",
        "Display": "Race Number",
        "Ops": [
          "eq",
          "ne",
          "gt",
          "ge",
          "lt",
          "le"
        ],
        "KeyTable": ""
      },
      {
        "Name": "race_round",
        "Display": "Race Round",
        "Ops": [
          "eq",
          "ne",
          "gt",
          "ge",
          "lt",
          "le"
        ],
        "KeyTable": ""
      },
      {
        "Name": "race_start",
        "Display": "Race Start",
        "Ops": [
          "eq",
          "ne",
          "gt",
          "ge",
          "lt",
          "le",
          "like"
        ],
        "KeyTable": ""
      }
    ]
  },
  {
    "Name": "org.jimmc.jraceman.RaceScore",
    "Display": "Race Results with Score",
    "Description": "Results for a Race, sorted by finish order and including score points",
    "OrderBy": [
      {
        "Name": "event",
        "Display": "Event, Race"
      },
      {
        "Name": "race",
        "Display": "Race"
      }
    ],
    "Where": [
      {
        "Name": "event_id",
        "Display": "Event",
        "Ops": [
          "eq"
        ],
        "KeyTable": "event"
      },
      {
        "Name": "event_name",
        "Display": "Event Name",
        "Ops": [
          "eq",
          "ne",
          "gt",
          "ge",
          "lt",
          "le",
          "like"
        ],
        "KeyTable": ""
      },
      {
        "Name": "event_number",
        "Display": "Event Number",
        "Ops": [
          "eq",
          "ne",
          "gt",
          "ge",
          "lt",
          "le"
        ],
        "KeyTable": ""
      },
      {
        "Name": "race_id",
        "Display": "Race",
        "Ops": [
          "eq"
        ],
        "KeyTable": "race"
      },
      {
        "Name": "race_number",
        "Display": "Race Number",
        "Ops": [
          "eq",
          "ne",
          "gt",
          "ge",
          "lt",
          "le"
        ],
        "KeyTable": ""
      },
      {
        "Name": "race_round",
        "Display": "Race Round",
        "Ops": [
          "eq",
          "ne",
          "gt",
          "ge",
          "lt",
          "le"
        ],
        "KeyTable": ""
      },
      {
        "Name": "race_start",
        "Display": "Race Start",
        "Ops": [
          "eq",
          "ne",
          "gt",
          "ge",
          "lt",
          "le",
          "like"
        ],
        "KeyTable": ""
      }
    ]
  }
]